adafruit_pybadger/pyportal.py

```diff
@@ -0,0 +1,72 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2020 Kattni Rembor for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_pybadger.pyportal`
+================================================================================
+
+Badge-focused CircuitPython helper library for PyPortal.
+
+
+* Author(s): Kattni Rembor
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+* `Adafruit PyPortal <https://www.adafruit.com/product/4116>`_
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+
+import audioio
+from adafruit_pybadger.pybadger_base import PyBadgerBase
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_PyBadger.git"
+
+
+class PyPortal(PyBadgerBase):
+    """Class that represents a single PyPortal."""
+
+    _audio_out = audioio.AudioOut
+    _neopixel_count = 1
+
+    @property
+    def _unsupported(self):
+        """This feature is not supported on PyPortal."""
+        raise NotImplementedError("This feature is not supported on PyPortal.")
+
+    # The following is a list of the features available in other PyBadger modules but
+    # not available for PyPortal. If called while using a PyPortal, they will result in the
+    # NotImplementedError raised in the property above.
+    play_file = _unsupported
+    light = _unsupported
```

Comment:
They PyPortal also has a light sensor. Can't it be supported?

Reply:
Weird. The following line does dim down the screen. I've tested it in a pybadge and the behavior seems the same. 

pybadger.auto_dim_display()

Note: The dim code acts weird by itself. If you change the environment of *both* boards, they don't seem to react to it. It looks like it only changes the backlight once, even if it's on the loops. But seems like something for another bug request.

Reply:
Yes I think we can support the light sensor. I will work on adding that today. 

I think the auto_dim functionality uses the accelerometer in the PyBadge to check for movement and dim the display when there hasn't been any. The PyPortals do not have one so we won't be able to have it work the same way, but perhaps could be adapted to use touch or something else.